sqlglot/parser.py

```diff
@@ -2973,10 +2974,11 @@ def _parse_select(
         # duckdb supports leading with FROM x
         from_ = self._parse_from() if self._match(TokenType.FROM, advance=False) else None
 
+        hint = self._parse_hint()
```

Comment:
Why was this added? It looks like this affects the Postgres test. Is there documentation that backs this up?

Reply:
Ditto about the hint tokenization.

Let's assume we have the following query:
```SQL
/*+ comment */ SELECT ...
```

Before, this was treated as a comment so it'd be attached to the following `exp.Select` node, but now this'd be tokenized as a `TokenType.HINT`, so `exp.Select` needs to parse that before `TokenType.SELECT`

Reply:
I replied to the other thread. I don't think this is right? Hints should be after the `SELECT` token IIRC.